src/query/plan/rule_based_planner.hpp

```diff
@@ -654,6 +717,21 @@ class RuleBasedPlanner {
 
     return operators;
   }
+
+  std::unique_ptr<LogicalOperator> MergeWithCombinator(std::unique_ptr<LogicalOperator> curr_op,
+                                                       std::unique_ptr<LogicalOperator> last_op,
+                                                       const Tree &combinator) {
+    if (const auto *union_ = utils::Downcast<const CypherUnion>(&combinator)) {
+      return impl::GenUnion(*union_, std::move(last_op), std::move(curr_op), *context_->symbol_table);
+    }
+
+    throw utils::NotYetImplemented("query combinator");
```

Comment:
More informative exception message.

Reply:
It wasn't written by me, I'm going to think of something appropriate